pkg/datapath/linux/node.go

```diff
@@ -1049,12 +1049,12 @@ func (n *linuxNodeHandler) NodeValidateImplementation(nodeToValidate node.Node)
 	return n.nodeUpdate(nil, &nodeToValidate, false)
 }
 
-// NodeDeviceNameWithDefaultRoute returns the node's device name which
+// NodeDeviceLinkWithDefaultRoute returns the node's device link which
 // handles the default route in the current namespace
-func NodeDeviceNameWithDefaultRoute() (string, error) {
+func NodeDeviceLinkWithDefaultRoute() (netlink.Link, error) {
 	link, err := route.NodeDeviceWithDefaultRoute()
 	if err != nil {
-		return "", err
+		return nil, err
 	}
-	return link.Attrs().Name, nil
+	return link, nil
```

Comment:
This is now exactly the same as `route.NodeDeviceWithDefaultRoute()`? Why not use the latter?

Reply:
Sure it doesn't really do much to bounce it through a call like this.